README.md

```diff
@@ -2,13 +2,160 @@
+
+Think of it kind of as though every `dispatch` in your app had been magically
+turned into `dispatch-sync`, and re-frame had lifted the restriction that says
+you can't call `dispatch-sync` from within an event handler.
+
+This macro is applicable for most events that do not run async behaviour within the 
+event.
+
+From the todomvc example:
+
+    (defn test-fixtures
+      []
+      ;; change this coeffect to make tests start with nothing
+      (rf/reg-cofx
+        :local-store-todos
+        (fn [cofx _]
+          "Read in todos from localstore, and process into a map we can merge into app-db."
+          (assoc cofx :local-store-todos
+                      (sorted-map)))))
+
+Define some test-fixtures in this case we have to ignore the localstore
+in the tests.
+
+    (deftest basic--sync
+      (rf-test/run-test-sync
+        (test-fixtures)
+        (rf/dispatch [:initialise-db])
+        
+Use the `run-test-sync` macro to construct the tests and initialise the app state
+note that the `dispatch` will be handled before the following code is executed, 
+effectively turning it into a `dispatch-sync`. Also any changes to the database
+and registrations will be rolled back at the termination of the test, therefore 
+our fixtures are run within the `run-test-sync` macro.
+    
+        (let [showing         (rf/subscribe [:showing])
+              sorted-todos    (rf/subscribe [:sorted-todos])
+              todos           (rf/subscribe [:todos])
+              visible-todos   (rf/subscribe [:visible-todos])
+              all-complete?   (rf/subscribe [:all-complete?])
+              completed-count (rf/subscribe [:completed-count])
+              footer-counts   (rf/subscribe [:footer-counts])]
+              
+Define subscriptions to the app state
+
+          (is (= :all @showing))
+          (is (empty? @sorted-todos))
+          (is (empty? @todos))
+          (is (empty? @visible-todos))
+          (is (= false (boolean @all-complete?)))
+          (is (= 0 @completed-count))
+          (is (= [0 0] @footer-counts))
+          
+Assert the initial state
+    
+          (rf/dispatch [:add-todo "write first test"])
+          
+Dispatch the event that you want to test, remember that `re-frame-test` will process
+this event immediately.
+
+          (is (= 1 (count @todos) (count @visible-todos) (count @sorted-todos)))
+          (is (= 0 @completed-count))
+          (is (= [1 0] @footer-counts))
+          (is (= {:id 1, :title "write first test", :done false}
+                 (first @todos)))
+                 
+Test that the dispatch has mutated the state in the way that we expect.
+    
+### run-test-async
+Run `body` as an async re-frame test. The async nature means you'll need to
+use `wait-for` any time you want to make any assertions that should be true
+*after* an event has been handled.  It's assumed that there will be at least
+one `wait-for` in the body of your test (otherwise you don't need this macro
+at all).
```

Comment:
You should mention that it is presumed that the event that is caught by `wait-for` is a response from an asynchronous event.